Docs: File transfer through the WebUI and Teleport Connect (#41798)
gravitational/teleport: docs/pages/connect-your-client/introduction.mdx

</TabItem>
</Tabs>

### Transferring files

<Tabs>
<TabItem label="tsh scp">

`tsh scp` will let you transfer files to servers behind Teleport:

```code
$ tsh scp some-file.ext server.example.com:
some-file.ext   7% |███████                | (25/342 MB, 2.9 MB/s) [9s:1m48s]
```
</TabItem>
<TabItem label="Teleport Connect">

Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows:

![Download with connect](../../img/download-with-connect.png)

You can upload and download files from systems using drag and drop:

> ![Drag and drop in WebUI](../../img/teleport-connect-drag-drop.png)

Reviewer: ```suggestion
![Drag and drop in Connect](../../img/teleport-connect-drag-drop.png)
```

</TabItem>
<TabItem label="Web UI">

> You can also use the arrows in the top-left corner of the Web UI to upload and download files:

Reviewer: ```suggestion
You can use the arrows in the top-left corner of the Web UI to upload and download files:
```

It'd be worth to mention that in the tab for Connect as well, since the UI for accessing file transfers is pretty much the same.

</TabItem>
</Tabs>

### Transferring files

<Tabs>
<TabItem label="tsh scp">

`tsh scp` will let you transfer files to servers behind Teleport:

```code
$ tsh scp some-file.ext server.example.com:
some-file.ext   7% |███████                | (25/342 MB, 2.9 MB/s) [9s:1m48s]
```
</TabItem>
<TabItem label="Teleport Connect">

> Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows. You can use the arrows in the top-left corner of the Web UI to upload and download files:

Reviewer: ```suggestion
Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows. You can use the arrows in the top-left corner of an active SSH session to upload and download files:
```

Author: ✅

> ![Download with connect](../../img/download_files_connect.png)

Reviewer 2: We'll need to update the image reference here

Author: I have been saving these in the `img` directory and they're all resolving for me:

```
-rw-r--r--@  1 mmcallister  staff    18024 May 22 12:20 upload_files_webui.png
-rw-r--r--@  1 mmcallister  staff    12826 May 22 12:21 download_files_webui.png
-rw-r--r--@  1 mmcallister  staff    50658 May 23 11:07 download_files_connect.png
-rw-r--r--@  1 mmcallister  staff    59520 May 23 11:53 upload_files_connect_drag_drop.png
/Users/mmcallister/Teleport/teleport/docs/img [copy_files_guide]
```
